yarn-project/foundation/src/json-rpc/convert.ts

```diff
@@ -75,7 +75,10 @@ export function convertFromJsonObj(cc: ClassConverter, obj: any): any {
     }
     return newObj;
   }
-
+  // Throw if this is a non-primitive class that was not registered
+  if (typeof obj === 'object' && Object.getPrototypeOf(obj) === Object.getPrototypeOf({})) {
```

Comment:
doesn't this return true for primitive objects?

Reply:
think you meant to write `typeof obj === 'object' && Object.getPrototypeOf(obj) !== Object.getPrototypeOf({})`?

Reply:
You're absolutely right